BSP/Simulator/DataGeneration/simulate.py

```diff
@@ -213,18 +213,32 @@ def main():
                     print("Enter the frequency you would like to change the clock to in Hz.")
                     frequency = int(input())
                     PLL.Change_Frequency(frequency)
+                elif choice == 'CAN':
+                    print("Enter the CAN ID for the system you wish to simulate. Leave out '0x'.")
+                    id = input()
+                    while(CAN.Invalid_CAN_ID(id) == True):
+                        print("Invalid CAN ID. Try again.")
+                        id = input()
+                    print("Enter up to 8 bytes of the CAN message that you would like to send, and separate each byte by a ','. Leave out '0x'.")
+                    message = input().split(',')
+                    CAN.Send_Message(id, message, len(message))
                 else:
                     print("That is not a valid option. Continuing simulation...")
                     stdscr = curses.initscr()
                     curses.start_color()
```

Comment:
So at this point, everything from line 210 to 228 is exactly the same as what is repeated below it. I think it would probably be better to restructure this so that we don't have to repeat code as much. Maybe something like this.

```python
if BeVolt is not None:
    print('wires, PLL, CAN, quit')
else:
    print('config, PLL, CAN, quit')
print('>>')

if choice == 'wires' and BeVolt is not None:
    # Do stuff
elif choice == 'quit':
    # ...
elif choice == 'PLL':
    # ...
elif choice == 'CAN':
    # ...
elif choice == 'config' and BeVolt is None:
    # ...
else:
    # invalid
```